C/plugins/south/dht11/dht11.cpp

```diff
@@ -0,0 +1,146 @@
+/*
+ * FogLAMP south service plugin
+ *
+ * Copyright (c) 2018 OSIsoft, LLC
+ *
+ * Released under the Apache 2.0 Licence
+ *
+ * Author: Amandeep Singh Arora
+ */
+#include <sstream>
+#include <iostream>
+#include <reading.h>
+#include <wiringPi.h>
+#include <dht11.h>
+#include <logger.h>
+
+#define MAX_SENSOR_READ_RETRIES		(100)
+
+using namespace std;
+
+/**
+ * Constructor for the dht11 "sensor"
+ *
+ * @param pin	The RPi pin to which DHT11 sensor's data pin is attached (e.g. pin 7 = GPIO 4, this is the default)
+ */
+DHT11::DHT11(unsigned int pin) :
+m_pin(pin)
+{
+	if ( wiringPiSetup() == -1 ) {
+		Logger::getLogger()->fatal("wiringPiSetup failed for dht11, can't proceed");
+		throw runtime_error("wiringPiSetup failed, can't proceed");
+	}
+	(void) takeReading(true);
+}
+
+/**
+ * Destructor for the dht11 "sensor"
+ */
+DHT11::~DHT11()
+{
+}
+
+/**
+ * Attempt to read temperature and humidity data from sensor
+ *
+ * @param sensorData   This buffer is filled with sensor data if read succeeds & data checksum matches
+ */
+bool DHT11::readSensorData(uint8_t sensorData[])
+{
+        uint8_t laststate       = HIGH;
+        uint8_t counter         = 0;
+        uint8_t j               = 0, i;
+        float   f; /* fahrenheit */
+	uint8_t dht11_dat[5] = {0, 0, 0, 0, 0};
+
+        /* pull pin down for 18 milliseconds */
+        pinMode( m_pin, OUTPUT );
+        digitalWrite( m_pin, LOW );
+        delay( 18 );
```

Comment:
this is standard? or should be configurable?

Reply:
This is standard.